src/App/App.js

```diff
@@ -90,6 +92,21 @@ const App = () => {
         };
     }, []);
     React.useEffect(() => {
+        const onCoreEvent = ({ event, args }) => {
+            switch (event) {
+                case 'SettingsUpdated': {
+                    if (args && args.settings && typeof args.settings.interfaceLanguage === 'string') {
+                        i18n.changeLanguage(args.settings.interfaceLanguage);
+                    }
```

Comment:
You can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to simplify this. All latest browsers except IE already [support](https://caniuse.com/?search=optional%20chain) it and it's included in babel's `@babel/preset-env` preset.

```suggestion
                   const language = args?.settings?.interfaceLanguage;
                    if (typeof language === 'string') {
                        i18n.changeLanguage(language);
                    }
```
The additional const variable assignment in my suggestion is optional, but makes it even more readable imo.

Reply:
I did not used optional chaining because i wanted to follow the current way everything is checked in the app
Also i think in js it might be clearer to spot mistakes with this syntax since there are no linting

Reply:
let's indeed keep it consistent and update it all at once if the time arises

Reply:
There are other places where optional chaining is used. So an inconsistency already exists. I agree this is out of scope for this PR though, we can pick one style and create a refactoring PR later.